src/std-traits/casting.md

```diff
@@ -20,6 +20,11 @@ The results of `as` are _always_ defined in Rust and consistent across
 platforms.  This might not match your intuition for changing sign or casting to
 a smaller type -- check the docs, and comment for clarity.
 
+Casting with `as` is a relatively sharp tool that is easy to use incorrectly, and can be a source of subtle bugs as future maintenance work changes the types that are used or the ranges of values in types. Casts are best used only when the intent is to indicate unconditional truncation (e.g. selecting the bottom 32 bits of a `u64` with `as u32`, regardless of what was in the high bits).
+
+For infallible casts (e.g. `u32` to `u64`), prefer using `From` or `Into` over `as`
+to confirm that the cast is in fact infallible. For fallible casts, `TryFrom` and `TryInto` are available when you want to handle casts that fit differently from those that don't.
+
```

Comment:
This was the one bit of content that seemed to be missing detail to me, so I fleshed it out with guidance derived from what my team has found to work well doing a fair bit of bit wrangling.

Side note: I expected `dprint fmt` to wrap this, but it didn't 🤷

Reply:
> Side note: I expected `dprint fmt` to wrap this, but it didn't 🤷

This is because we've been excluding the `src/` directory until now. I have #1157 up for this, but I was waiting for the v2 rewrite to land first. I will re-fresh that PR now and then we can have everything consistent.